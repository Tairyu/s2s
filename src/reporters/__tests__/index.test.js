// @flow
import chalk from 'chalk'
import stripAnsi from 'strip-ansi'

jest.resetModules()
jest.doMock('chalk', () => new chalk.constructor({ enabled: true }))

// eslint-disable-next-line
import { formatText, trimAndFormatPath } from '../'

test('trimAndFormatPath', () => {
  expect(trimAndFormatPath('/src/user/hoge/index.js')).toMatchSnapshot()
})

test('formatText when handlerType = s2s', () => {
  const inputPath = 'src/input/input.js'
  const outputPath = 'src/output/output.js'
  const result = formatText('S2S', inputPath, outputPath)
  expect(result).toMatchSnapshot()
  expect(stripAnsi(result)).toMatchSnapshot()
})

test('formatText when handlerType = template', () => {
  const inputPath = 'src/input/input.js'
  const outputPath = 'src/output/output.js'
  const result = formatText('TEMPLATE', inputPath, outputPath)
  expect(result).toMatchSnapshot()
  expect(stripAnsi(result)).toMatchSnapshot()
})
