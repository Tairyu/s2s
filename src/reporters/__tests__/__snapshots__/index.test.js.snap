// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatText when handlerType = s2s 1`] = `"</><bold><green>S2S</></> <dim>src/input/<bold>input.js → <dim>src/output/<bold>output.js"`;

exports[`formatText when handlerType = s2s 2`] = `"S2S src/input/input.js → src/output/output.js"`;

exports[`formatText when handlerType = template 1`] = `"</><bold><cyan>TEMPLATE</></> <dim>src/input/<bold>input.js → <dim>src/output/<bold>output.js"`;

exports[`formatText when handlerType = template 2`] = `"TEMPLATE src/input/input.js → src/output/output.js"`;

exports[`trimAndFormatPath 1`] = `"<dim>/src/user/hoge/<bold>index.js"`;
